pkg/storage/allocator.go

```diff
@@ -534,11 +534,8 @@ func (a Allocator) RebalanceTarget(
 			ReplicaID: rangeInfo.Desc.NextReplicaID,
 		}
 
-		// Deep-copy the Replicas slice since we'll mutate it.
 		desc := *rangeInfo.Desc
-		desc.Replicas = append([]roachpb.ReplicaDescriptor(nil), desc.Replicas...)
-
-		desc.Replicas = append(desc.Replicas, newReplica)
+		desc.Replicas = append(desc.Replicas[:len(desc.Replicas):len(desc.Replicas)], newReplica)
```

Comment:
Please restore the comment. This code is nice and compressed but pretty sure most people will need to squint at it for a minute or two to figure out what's going on.

Reply:
Restored.